### ISSUE TITLE
Privado: custom endpoint port is not allowed

### ISSUE THREAD

**Issue by Qballjos** [qdm12/gluetun#2347]
### Discussed in https://github.com/qdm12/gluetun/discussions/2346

<div type='discussions-op-text'>

<sup>Originally posted by **Qballjos** July  5, 2024</sup>
unable to setup this container with provider Privado selected, not with the template or docker compose.
It seems that it is stuck trying to use port 51820
also used custom config with privado settings and manualy changing the hostname to an ip still not picking up the wanted port 1194</div>

**Comment by Amlor** [comment]
Yep, have the same error with protonvpn after image update

**Comment by qdm12** [comment]
@Qballjos I'm not sure I follow. Privado only supports Openvpn UDP on port 1194 which is the default. Running for example `docker run -it --rm --cap-add=NET_ADMIN -e VPN_SERVICE_PROVIDER=privado -e OPENVPN_USER=a -e OPENVPN_PASSWORD=b  qmcgaw/gluetun` tries to connect on port `1194`, for example it logs out: `[openvpn] TCP/UDP: Preserving recently used remote address: [AF_INET]91.148.224.20:1194` so that seems correct to me.

@Amlor Please create an issue with required fields information and details, since I really cannot help you without additional information, and this issue is about Privado.

**Comment by TheRealBix** [comment]
Hi, I think I have the same issue this person tried to report to.
I was previously using wireguard and now i'm trying to switch to openvpn, but "classic" config will result in error.

Here's my compose :

```
  gluetun:
    image: qmcgaw/gluetun:latest
    container_name: gluetun
    cap_add:
      - NET_ADMIN
    devices:
      - /dev/net/tun:/dev/net/tun
    ports:
      - 9865:9865/tcp
    volumes:
      - /volume2/docker/gluetun:/gluetun
      # - /volume2/docker/gluetun/wireguard/wg0.conf:/gluetun/wireguard/wg0.conf
    environment:
      - PUID=1029
      - PGID=100
      - TZ=Europe/Paris
      # - VPN_SERVICE_PROVIDER=custom                       #config wireguard
      # - VPN_TYPE=wireguard                                #config wireguard
      # - VPN_PORT_FORWARDING_PROVIDER=protonvpn            #config wireguard
      - VPN_SERVICE_PROVIDER=protonvpn                                #config openvpn
      - VPN_PORT_FORWARDING_USER=sdfsdfsdfs+pmp            #config openvpn
      - VPN_PORT_FORWARDING_PASSWORD=sdfsdfsdfsdfsdf   #config openvpn
      - VPN_PORT_FORWARDING=on                                        #config openvpn
      - SERVER_CITIES=paris                                           #config openvpn
      - FIREWALL_OUTBOUND_SUBNETS=172.17.0.0/16,192.168.1.0/24
      - UPDATER_PERIOD=24h
      - BLOCK_MALICIOUS=off
      - BLOCK_SURVEILLANCE=off
      - BLOCK_ADS=off
    network_mode: bridge
    labels:
      - com.centurylinklabs.watchtower.enable=false
    security_opt:
      - no-new-privileges:true
    restart: unless-stopped
```

Here's some log :

```
Running version latest built on 2024-07-12T19:57:02.146Z (commit 9d50c23)

🔧 Need help? https://github.com/qdm12/gluetun/discussions/new
🐛 Bug? https://github.com/qdm12/gluetun/issues/new
✨ New feature? https://github.com/qdm12/gluetun/issues/new
☕ Discussion? https://github.com/qdm12/gluetun/discussions/new
💻 Email? quentin.mcgaw@gmail.com
💰 Help me? https://www.paypal.me/qmcgaw https://github.com/sponsors/qdm12
2024-07-26T11:38:17+02:00 INFO [routing] default route found: interface eth0, gateway 172.17.0.1, assigned IP 172.17.0.4 and family v4
2024-07-26T11:38:17+02:00 INFO [routing] local ethernet link found: eth0
2024-07-26T11:38:17+02:00 INFO [routing] local ipnet found: 172.17.0.0/16
2024-07-26T11:38:17+02:00 INFO [firewall] enabling...
2024-07-26T11:38:17+02:00 INFO [firewall] enabled successfully
2024-07-26T11:38:19+02:00 INFO [storage] merging by most recent 19425 hardcoded servers and 19425 servers read from /gluetun/servers.json
2024-07-26T11:38:19+02:00 ERROR VPN settings: provider settings: server selection: OpenVPN server selection settings: custom endpoint port is not allowed: for VPN service provider protonvpn: value is not one of the possible choices: 51820 must be one of 80, 443, 1194, 4569 or 5060
2024-07-26T11:38:19+02:00 INFO Shutdown successful
========================================
========================================
=============== gluetun ================
========================================
=========== Made with ❤️ by ============
======= https://github.com/qdm12 =======
========================================
========================================

Running version latest built on 2024-07-12T19:57:02.146Z (commit 9d50c23)

🔧 Need help? https://github.com/qdm12/gluetun/discussions/new
🐛 Bug? https://github.com/qdm12/gluetun/issues/new
✨ New feature? https://github.com/qdm12/gluetun/issues/new
☕ Discussion? https://github.com/qdm12/gluetun/discussions/new
💻 Email? quentin.mcgaw@gmail.com
💰 Help me? https://www.paypal.me/qmcgaw https://github.com/sponsors/qdm12
2024-07-26T11:39:19+02:00 INFO [routing] default route found: interface eth0, gateway 172.17.0.1, assigned IP 172.17.0.4 and family v4
2024-07-26T11:39:19+02:00 INFO [routing] local ethernet link found: eth0
2024-07-26T11:39:19+02:00 INFO [routing] local ipnet found: 172.17.0.0/16
2024-07-26T11:39:19+02:00 INFO [firewall] enabling...
2024-07-26T11:39:19+02:00 INFO [firewall] enabled successfully
2024-07-26T11:39:20+02:00 INFO [storage] merging by most recent 19425 hardcoded servers and 19425 servers read from /gluetun/servers.json
2024-07-26T11:39:21+02:00 ERROR VPN settings: provider settings: server selection: OpenVPN server selection settings: custom endpoint port is not allowed: for VPN service provider protonvpn: value is not one of the possible choices: 51820 must be one of 80, 443, 1194, 4569 or 5060
2024-07-26T11:39:21+02:00 INFO Shutdown successful
========================================
========================================
=============== gluetun ================
========================================
=========== Made with ❤️ by ============
======= https://github.com/qdm12 =======
========================================
========================================

Running version latest built on 2024-07-12T19:57:02.146Z (commit 9d50c23)

🔧 Need help? https://github.com/qdm12/gluetun/discussions/new
🐛 Bug? https://github.com/qdm12/gluetun/issues/new
✨ New feature? https://github.com/qdm12/gluetun/issues/new
☕ Discussion? https://github.com/qdm12/gluetun/discussions/new
💻 Email? quentin.mcgaw@gmail.com
💰 Help me? https://www.paypal.me/qmcgaw https://github.com/sponsors/qdm12
2024-07-26T11:40:21+02:00 INFO [routing] default route found: interface eth0, gateway 172.17.0.1, assigned IP 172.17.0.4 and family v4
2024-07-26T11:40:21+02:00 INFO [routing] local ethernet link found: eth0
2024-07-26T11:40:21+02:00 INFO [routing] local ipnet found: 172.17.0.0/16
2024-07-26T11:40:21+02:00 INFO [firewall] enabling...
2024-07-26T11:40:21+02:00 INFO [firewall] enabled successfully
2024-07-26T11:40:22+02:00 INFO [storage] merging by most recent 19425 hardcoded servers and 19425 servers read from /gluetun/servers.json
2024-07-26T11:40:23+02:00 ERROR VPN settings: provider settings: server selection: OpenVPN server selection settings: custom endpoint port is not allowed: for VPN service provider protonvpn: value is not one of the possible choices: 51820 must be one of 80, 443, 1194, 4569 or 5060
2024-07-26T11:40:23+02:00 INFO Shutdown successful

```

**Comment by TheRealBix** [comment]
Gluetun seems to keep trying to read config files instead of just following env settings.

I modified the "wg0.conf" to "wg0.conf.old", which gluetun shouldn't use as it's configured to use openvpn and now gluetun say this : 
`ERROR VPN settings: OpenVPN settings: user is empty`

To my understanding, with my env settings, gluetun shouldn't read any config file...

**Comment by qdm12** [comment]
This bug was introduced in https://github.com/qdm12/gluetun/commit/ecc80a5a9e3f5ba8c3096eb47c9ed8544a7e8867 after the v3.38.0 release. It's fixed in 36c8da7ea7152d71ba62c5f1884784793d4facde which, as a side effect, also changes, with retro-compatibility:

- Split `VPN_ENDPOINT_IP` in `OPENVPN_ENDPOINT_IP` and `WIREGUARD_ENDPOINT_IP`
- Split `VPN_ENDPOINT_PORT` in `OPENVPN_ENDPOINT_PORT` and `WIREGUARD_ENDPOINT_PORT`

This (retro-compatible) change will be documented in the wiki soon ™️
